File: app/src/main/java/im/point/dotty/db/CommentDao.java
package im.point.dotty.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import im.point.dotty.model.Comment;
import io.reactivex.Observable;

@Dao
public interface CommentDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addComments(Comment... comments);

    @Query("SELECT * FROM comments WHERE post_id = :postId ORDER BY timestamp")
    Observable<List<Comment>> getComments(String postId);
}

File: app/src/main/java/im/point/dotty/db/CommentDao.kt
/*
 * Copyright (c) 2019-2021 Pavel A. Sokolov
 */
package im.point.dotty.db

import androidx.room.Dao
import androidx.room.Query
import im.point.dotty.model.Comment
import kotlinx.coroutines.flow.Flow

@Dao
interface CommentDao : CommonDao<Comment> {
    @Query("SELECT * FROM comments WHERE post_id = :id ")
    fun getItem(id: String): Comment?

    @Query("SELECT * FROM comments WHERE id = :id")
    fun geItemFlow(id: String): Flow<Comment?>

    @Query("SELECT * FROM comments WHERE post_id = :id")
    fun getPostComments(id: String): List<Comment>

    @Query("SELECT * FROM comments WHERE post_id = :id")
    fun getPostCommentsFlow(id: String): Flow<List<Comment>>
}